source/blog/2017-02-28-rdo-ocata-release-behind-the-scenes.html.md

```diff
@@ -0,0 +1,62 @@
+---
+title: RDO Ocata Release Behind The Scenes
+author: hguemar
+date: 2017-02-28 08:00:00 UTC
+tags: openstack,ocata,release,releng,behind-the-scenes
+comments: true
+published: true
+---
+
+I have been involved in 6 GA releases of RDO (From Juno to Ocata), and I wanted to share a glimpse of the preparation work.
+Since Juno, our process has tremendously evolved: we refocused RDO on EL7, joined the CentOS Cloud SIG, moved to [Software Factory](https://www.redhat.com/archives/rdo-list/2015-October/msg00123.html).
+
+Our release process does not start when upstream announces GA or even a milestone, no it starts from the very beginning of upstream cycle.
+
+**Trunk chasing**
+
+We have been using [DLRN](https://www.rdoproject.org/what/dlrn/) to track upstream changes and build continuously OpenStack as a RPM distribution.
+Then our [CI](https://ci.centos.org/view/rdo/view/promotion-pipeline/) hosted on [CentOS community CI](https://wiki.centos.org/QaWiki/CI) runs multiple jobs
+on DLRN snapshots. We use the [WeIRDO](https://github.com/rdo-infra/weirdo) framework to run the same jobs as upstream CI on our packages.
+This allows us to detect early integration issues and get either our packaging or upstream projects fixed. This also includes installers such as OPM, TripleO or PackStack.
+
+We also create Ocata tags in CentOS Community Build System (CBS) in order to build dependencies that are incompatible with currently supported releases.
+
+
+**Branching**
+
+We start branching RDO stable release around milestone 3, and have stable builds getting bootstrapped. This includes:
+
+* registering packages in CBS, I scripted this part for Ocata using rdoinfo database.
```

Comment:
Dependencies synchronization with upstream changes during the development cycle. I think we can consider it part of the branching process.